### ISSUE TITLE
[Bug] --serve flag does not work (docfx: 2.66)

### ISSUE THREAD

**Issue by chrisdecker1201** [dotnet/docfx#8681]
**Describe the bug**
The `--serve` flag does not work in 2.66

**To Reproduce**
~~~bash
dotnet tool update -g docfx --version 2.66
mkdir test
cd test
docfx init --quiet
docfx docfx_project/docfx.json --serve
~~~

The documentation is builded, but no endpoint is started.

**Expected behavior**
~~~bash
dotnet uninstall -g docfx
dotnet tool update -g docfx --version 2.65
mkdir test
cd test
docfx init --quiet
docfx docfx_project/docfx.json --serve
~~~

The documentation is builded and I can visit it on http://localhost:8080/

**Context (please complete the following information):**
- OS: Windows 10
- Docfx version: 2.66

**Comment by yufeih** [comment]
> `docfx build --serve` and `docfx serve` work in DocFX 2.66.0, but `docfx --serve` without a subcommand seems to ignore `--serve`.

Correct. The `docfx --serve` command used to bindly pass all args to each individual command, causing problems with commands that do not know the args, such as the pdf command.

We can enable this on a per args basis though, it seems reasonable to have `--serve`.
